### PR TITLE
Fix SITL Test failure: Place PX4 instance runner after Gazebo server runner

### PR DESCRIPTION
**Pull request by junwoo091400**
## About
- This was a nasty bug where starting PX4 instance first, then starting Gazebo server was causing PX4 instance' EKF to freak out, probably because it doesn't like getting data a while after it is started
- This now guarantees that such edge case won't occur and MAVSDK test will run as it should

For detailed troubleshooting log, please read: https://github.com/PX4/PX4-Autopilot/issues/21229

### Solved Problem
Fixes https://github.com/PX4/PX4-Autopilot/issues/21229

### Solution
Start PX4 instance AFTER Gazebo server is started

### Alternatives
We could also make px4 itself robust so that it doesn't freak out when a simulated sensor data starts flowing in after it is instantiated for a while, but that's up for a discussion.

### Test coverage
All 3 out of 4 tests now always pass consistently:
image

NOTE: VTOL Mission Test (4/4) failure isn't related to MAVSDK Test architecture. Reason is given here: https://github.com/PX4/PX4-Autopilot/issues/21229#issuecomment-1450792550

### Comment
I think there's a high chance that this is the root cause of the https://github.com/PX4/PX4-Autopilot/issues/14835 as well.

### CODE REVIEW

**Comment by junwoo091400** [comment]
This is a minimal change & I have observed that it does solve the root cause of px4 instance hanging. @dagar could you quickly review this so we can get it in?

**Comment by junwoo091400** [comment]
I will rebase to take into account the fix PR for tailsitter: https://github.com/PX4/PX4-Autopilot/pull/21319

Maybe we will have the first *consistent green checkmark on Tailsitter SITL in ages :raised_hands:

**Comment by junwoo091400** [comment]
Let's see if it passes!

File: test/mavsdk_tests/mavsdk_test_runner.py [hunk 1]
```diff
@@ -441,6 +430,25 @@ def start_runners(self,
                         self.verbose)
                     self.active_runners.append(gzclient_runner)
 
+                '''
+                WARNING: DO NOT PLACE PX4 RUNNER BEFORE GAZEBO RUNNER!
+                # We must start the PX4 instance at the end, as starting
+                it in the beginning, then connecting Gazebo server freaks
+                out the PX4 (it needs to have data coming in when started),
+                and can lead to EKF to freak out, or the instance itself
+                to die unexpectedly.
```
Comment (julianoes) [inline test/mavsdk_tests/mavsdk_test_runner.py:439]:
```suggestion
                # We must start the PX4 instance at the end, as starting
                # it in the beginning, then connecting Gazebo server freaks
                # out the PX4 (it needs to have data coming in when started),
                # and can lead to EKF to freak out, or the instance itself
                # to die unexpectedly.
```

In my opinion this shouldn't be that shouty.

**Comment by junwoo091400** [comment]
image

Ah no, the attitude failure & failsafe problem still persists.

**Comment by junwoo091400** [comment]
image

Now it seems to work!! Testing procedure:
```bash
make px4_sitl_default sitl_gazebo-classic mavsdk_tests

PX4_CMAKE_BUILD_TYPE=RelWithDebInfo test/mavsdk_tests/mavsdk_test_runner.py --speed-factor 100 --abort-early --model tailsitter test/mavsdk_tests/configs/sitl.json --verbose --case "Fly VTOL mission"
```

**Comment by junwoo091400** [comment]
Seems like in HITL simulations, when the `gps stop` is commanded the Flight Tasks go into failsafe mode.

image

What's weird is that this error only happens on certain build targets, which indicates that it isn't really the software problem :thinking:, since I am guessing all the targets will share mostly the same codebase, and error in flight task manager seems quite random? @dagar 

image

**Comment by junwoo091400** [comment]
Since I identified the commit that caused the HITL Jenkins rack testing error in https://github.com/PX4/PX4-Autopilot/issues/21370, this PR is good to merge as all the other CI has passed :tada:

**Comment by junwoo091400** [comment]
Could you approve this @julianoes ? Let's get the CI working great again :mechanical_arm:

**Review by julianoes — APPROVED** [review]

**Comment by junwoo091400** [comment]
Thanks!!
